app/general/admin.py

```diff
@@ -26,10 +25,14 @@ def clean(self):
         url = cleaned_data.get("url", "")
         uploaded_file = cleaned_data.get("uploaded_file", "")
 
-        if cleaned_data["mime_type"] is not None:
-            cleaned_data["mime_type"] = (
-                mimetypes.guess_type(uploaded_file.name)[0] if uploaded_file else ""
-            )
+        if uploaded_file:
+            file_type = magic.from_buffer(uploaded_file.read(), mime=True)
+            if file_type != "application/pdf":
+                self.add_error("uploaded_file", "Only PDF files are allowed.")
+
+            cleaned_data["mime_type"] = file_type if uploaded_file else ""
```

Comment:
The check for `if uploaded_file` is superfluous inside the `if uploaded_file` block. Should the assignment of MIME type maybe be removed from the if block to happen always? Just wondering if the MIME type will be blanked correctly if we remove the PDF file in a subsequent save.

Reply:
I went back and tested and double-checked and this line of code is very important as it saves the meme_type to the Database, the meme_type is also removed when the file upload is removed from the form.

Reply:
I agree that the line is important. My comment is about the superfluous `if uploaded_file else ""` part of the line if it is always inside a conditional block that means the first clause is always executed, and the `else` clause is never executed.

I'm happy if the mime_type field is cleared. I just can't see code that would effect that, since the user can't edit it, and `clean()` doesn't touch it if there is no uploaded file.